### PR TITLE
Fix LLVM 18 warnings and remove protobuf

### PR DESCRIPTION
**Pull request by no92**
This PR fixes new warnings emitted by LLVM 18. As a bonus, all remaining protobuf protocols have been converted to bragi, as protobuf was producing massive amounts of warnings.

### CODE REVIEW

File: tools/bakesvr/src/main.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#include <yaml-cpp/yaml.h>
+#include <svrctl.bragi.hpp>
+
+int main(int argc, char **argv) {
+	std::string input{};
+	std::string output{};
+
+	CLI::App app{"bakesvr: generate info for runsvr"};
+	app.add_option("input", input, "Path to the input file")->required();
+	app.add_option("-o,--output", output, "Path to the input file")->required();
```
Comment (Dennisbonke) [inline tools/bakesvr/src/main.cpp:15]: This should say path to the output file.

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
Quick glance showed me one error

File: hel/include/hel.h [hunk 2]
```diff
@@ -389,7 +389,7 @@ enum HelSyscallArgs {
 #endif
 
 struct HelQueueParameters {
-	uint32_t flags;
+	uint32_t flags = 0;
```
Comment (qookei) [inline hel/include/hel.h:392]: I don't think these default member initializers are correct. This header is supposed to be usable from both C and C++, and C doesn't support them.

File: drivers/usb/hcds/xhci/src/xhci.hpp [hunk 3]
```diff
@@ -136,7 +133,7 @@ struct Controller final : proto::BaseController {
 		uint8_t getSpeed();
 		int _id;
 		std::shared_ptr<Device> _device;
-		Controller *_controller;
```
Comment (qookei) [inline drivers/usb/hcds/xhci/src/xhci.hpp:139]: I think we can just remove this member completely instead.

File: tools/bakesvr/meson.build [hunk 4]
```diff
@@ -1,14 +1,10 @@
...
+	'src/main.cpp',
 )
 
-install_data('bakesvr', install_dir : get_option('bindir'))
+src += cxxbragi.process([protos/'svrctl/svrctl.bragi'])
+
+yaml_cpp_subproject = import('cmake').subproject('yaml-cpp')
+yaml_cpp_dep = yaml_cpp_subproject.dependency('yaml-cpp')
+
+executable('bakesvr', src, dependencies: [yaml_cpp_dep, cli11_dep, bragi_dep], install_dir : get_option('bindir'), native: true, install: true)
```
Comment (qookei) [inline tools/bakesvr/meson.build:10]: Could you reformat this? Spaces around `:` on both sides, and one option per line.

**Comment by no92** [comment]
Addressed the comments raised.

**Review by qookei — APPROVED** [review]

**Pull request merged**